x/tokenomics/keeper/settle_pending_claims.go

```diff
@@ -0,0 +1,147 @@
+			logger.Error(fmt.Sprintf("error settling session accounting for claim %s: %v", claim.SessionHeader.SessionId, err))
+			return 0, 0, err
+		}
+
+		claimSettledEvent := types.EventClaimSettled{
+			Claim:         &claim,
+			ComputeUnits:  claimComputeUnits,
+			ProofRequired: isProofRequiredForClaim,
+		}
+		if err := ctx.EventManager().EmitTypedEvent(&claimSettledEvent); err != nil {
+			return 0, 0, err
+		}
+
+		// The claim & proof are no longer necessary, so there's no need for them
+		// to take up on-chain space.
+		k.proofKeeper.RemoveClaim(ctx, sessionId, claim.SupplierAddress)
+		k.proofKeeper.RemoveProof(ctx, sessionId, claim.SupplierAddress)
+
+		numClaimsSettled++
+		logger.Info(fmt.Sprintf("Successfully settled claim %s at block height %d", claim.SessionHeader.SessionId, blockHeight))
+	}
+
+	logger.Info(fmt.Sprintf("settled %d claims and expired %d at block height %d", numClaimsSettled, numClaimsExpired, blockHeight))
+
+	return numClaimsSettled, numClaimsExpired, nil
+
+}
+
+// getExpiringClaims returns all claims that are expiring at the current block height.
+// This is the height at which the proof window closes.
+// If the proof window closes and a proof IS NOT required -> settle the claim.
+// If the proof window closes and a proof IS required -> only settle it if a proof is available.
+func (k Keeper) getExpiringClaims(ctx sdk.Context) (expiringClaims []prooftypes.Claim, err error) {
+	blockHeight := ctx.BlockHeight()
+
+	// TODO_BLOCKER: query the on-chain governance parameter once available.
+	// `* 3` is just a random factor Olshansky added for now to make sure expiration
+	// doesn't happen immediately after a session's grace period is complete.
+	submitProofWindowEndHeight := sessionkeeper.GetSessionGracePeriodBlockCount() * 3
+
+	// TODO_BLOCKER(@Olshansk): Optimize this by indexing claims appropriately
+	// and only retrieving the claims that need to be settled rather than all
+	// of them and iterating through them one by one.
+	claims := k.proofKeeper.GetAllClaims(ctx)
+
+	// Loop over all claims we need to check for expiration
+	for _, claim := range claims {
+		expirationHeight := claim.SessionHeader.SessionEndBlockHeight + submitProofWindowEndHeight
+		if blockHeight >= expirationHeight {
+			expiringClaims = append(expiringClaims, claim)
+		}
+	}
+
+	// Return the actually expiring claims
+	return expiringClaims, nil
+}
+
+// isProofRequiredForClaim checks if a proof is required for a claim.
+// If it is not, the claim will be settled without a proof.
+// If it is, the claim will only be settled if a valid proof is available.
+func (k Keeper) isProofRequiredForClaim(_ sdk.Context, claim *prooftypes.Claim) (bool, error) {
+	// NB: Assumption that claim is non-nil and has a valid root sum because it
+	// is retrieved from the store.
+	root := (smt.MerkleRoot)(claim.GetRootHash())
+	claimComputeUnits := root.Sum()
+	if claimComputeUnits < ProofRequiredComputeUnits {
```

Comment:
Does this mean that the `RelayMiner` would also use `ProofRequiredComputeUnits` to decide whether to submit a proof or not?

Doesn't this make it capable of creating fake claims if they do not exceed `ProofRequiredComputeUnits`?

It is planned to add that logic to the `RelayMiner`s session manager in this PR?